lib/src/line_scanner.dart

```diff
@@ -89,6 +89,13 @@ class LineScanner extends StringScanner {
     return character;
   }
 
+  @override
+  int readCodePoint() {
+    final codePoint = super.readCodePoint();
+    _adjustLineAndColumn(codePoint);
```

Comment:
Using `readCodePoint` to read a surrogate pair gives a different column index than two separate `readChar` calls.

Maybe that's what you want, maybe it isn't. We should at least document this:

> If you always use `readCodePoint`, your `column` is counted in code points. If you always use `readChar`, your `column` is counted in code units. If you mix them, don't use `column`.

The alternative is to always count code points, so `_adjustLineAndColumn` could be:
```dart
if (character <= 0xFFFF) { // or use  !inSupplementaryPlane(character) if speed is not important.
  ... current code ..
} else {
  _column += 2;
}
```

Whichever choice, make sure it's tested on `LineScanner` with mixed reads.

(And this is assuming we fix `super.readCodePoint` calling `readChar` twice.)

Reply:
Oops, this was just an oversight. This should definitely double-increment the column—that preserves the invariant that a `LineScanner`'s concept of "column" always matches a `SourceSpan`'s.

Reply:
That should work too.

Reply:
We have an open issue to change the concept of "column" in `source_span`.

I've started looking at the impact of that change but I haven't not gotten very far.
https://github.com/dart-lang/source_span/pull/84

Does that change impact how we would solve the problem in this package?